developer-portal/for-developers/custom-integration/fast-sdks/android/handle-fast-events.mdx

```diff
@@ -0,0 +1,153 @@
+| `FastCheckoutEvent.purchaseInfo` | `PurchaseInfo`      | Additional relevant data regarding the event. Refer to [PurchaseInfo Data Class Breakdown](#purchaseinfo-data-class-breakdown) for more details. |
+
+### FastCheckoutEvent Types
+
+The `FastCheckoutEvent` contains these events and associated data payloads.
+
+| Event            | Description                                                                                                                                                                                                                                     | Associated Data                               |
+| ---------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------- |
+| `OrderCreated`   | Emitted when the order has been successfully submitted through Fast Checkout.                                                                                                                                                                   | `PurchaseInfo`                                |
+| `OrderUpdated`   | Emitted when buyer updates order information like item quantity, removes an item, or adds a coupon.                                                                                                                                             | `PurchaseInfo`                                |
+| `OrderCompleted` | Emitted when order is finalized. This means that the batching window has ended and the user cannot update the order. If the user closes the SDK before batching window has ended, you won't get this event and instead get the SDKClosed event. | `PurchaseInfo`                                |
+| `OrderCanceled`  | Emitted when user cancels the order.                                                                                                                                                                                                            | `PurchaseInfo`                                |
+| `OrderFailed`    | Emitted when checkout fails with an error                                                                                                                                                                                                       | Throwable?                                    |
+| `SDKClosed`      | Buyer has exited the Fast Android SDK                                                                                                                                                                                                           | None Required (but can return `PurchaseInfo`) |
```

Comment:
I think we can update this line `None Required (but can return `PurchaseInfo`)` to `PurchaseInfo?`

Update this description to `Buyer has exited the Fast Android SDK` `Buyer has exited the Fast Android SDK. If buyer exits post purchase, we send purchase info with the event. We send `null` if buyer exits before clicking on CompletePurchase`

we can improve the language but this should get the message across.